homework/index.js

```diff
@@ -30,17 +34,96 @@
     return elem;
   }
 
+  function renderError(error) {
+    const root = document.getElementById('root');
+    root.innerHTML = '';
+    createAndAppend('h1', root, { text: error.message, class: `alert-error` });
+  }
+
+  function addRow(tbody, label, value) {
+    const tr = createAndAppend('tr', tbody);
+    createAndAppend('td', tr, { text: `${label}:`, class: 'label' });
+    createAndAppend('td', tr, { text: value });
+    return tr;
+  }
+
+  function renderRepos(repo, div) {
+    div.innerHTML = '';
+    const table = createAndAppend('table', div);
+    const tbody = createAndAppend('tbody', table);
+    const firstRow = addRow(tbody, 'Name', '');
+    createAndAppend('a', firstRow.lastChild, {
+      href: repo.html_url,
+      text: repo.name,
+      target: `_blank`,
+    });
+    addRow(tbody, 'Description', repo.description);
+    addRow(tbody, 'Forks', repo.forks);
+    addRow(tbody, 'Updated', new Date(repo.updated_at).toLocaleString('en-GB'));
+  }
+
+  function renderContributions(repo, ul) {
+    fetchJSON(repo.contributors_url, (err, contributors) => {
+      if (err) {
+        renderError(err);
+        return;
+      }
+      ul.innerHTML = '';
+      createAndAppend('h3', ul, { text: `Contributors:` });
+      contributors.forEach(contributor => {
+        const a = createAndAppend('a', ul, { href: contributor.html_url, target: '_blank' });
```

Comment:
I forgot this one: a `ul` can only have `li` children. The `a` tag needs to be child of the `li`, not the other way around.

Reply:
I put the a tag inside li tag